src/Backend/Modules/Pages/Actions/Edit.php

```diff
@@ -765,7 +765,17 @@ private function validateForm(): void
 
         $this->saveTags($page['id']);
 
-        BackendPagesModel::buildCache(BL::getWorkingLanguage());
+        $cacheShouldBeUpdated = !(
+            $this->record['title'] === $page['title']
+            && $this->record['navigation_title'] === $page['navigation_title']
+            && $this->record['navigation_title_overwrite'] === $page['navigation_title_overwrite']
+            && $this->record['hidden'] === $page['hidden']
+        );
+
+        // build cache
+        if ($cacheShouldBeUpdated) {
+            BackendPagesModel::buildCache(BL::getWorkingLanguage());
+        }
```

Comment:
This looks like something that could perfectly be done as a side effect using an event + listener.

Reply:
yup, sure :) I only made this PR because I had to do this for an older project. This should definitely be handled by an event listener/subscriber in the future.

Reply:
Might be a good moment to do this? 😄

Reply:
A good time for Fork, yes. A good time for me? No :p too much work unfortunately

Reply:
It's Hacktober fest @StijnVrolijk, don't you want a new t-shirt?